File: app/src/main/java/awais/instagrabber/models/FeedStoryModel.java
package awais.instagrabber.models;

import android.util.Log;

import androidx.annotation.NonNull;

import java.io.Serializable;
import java.util.Date;

import awais.instagrabber.utils.Utils;

public final class FeedStoryModel implements Serializable {
    private final String storyMediaId;
    private final ProfileModel profileModel;
    private final StoryModel firstStoryModel;
    private Boolean fullyRead;
    private final long timestamp;
    private final int mediaCount;

    public FeedStoryModel(final String storyMediaId, final ProfileModel profileModel, final boolean fullyRead,
                          final long timestamp, final StoryModel firstStoryModel, final int mediaCount) {
        this.storyMediaId = storyMediaId;
        this.profileModel = profileModel;
        this.fullyRead = fullyRead;
        this.timestamp = timestamp;
        this.firstStoryModel = firstStoryModel;
        this.mediaCount = mediaCount;
    }

    public String getStoryMediaId() {
        return storyMediaId;
    }

    public long getTimestamp() {
        return timestamp;
    }

    @NonNull
    public String getDateTime() {
        return Utils.datetimeParser.format(new Date(timestamp * 1000L));
    }

    public int getMediaCount() {
        return mediaCount;
    }

    public ProfileModel getProfileModel() {
        return profileModel;
    }

//    public void setFirstStoryModel(final StoryModel firstStoryModel) {
//        this.firstStoryModel = firstStoryModel;
//    }

    public StoryModel getFirstStoryModel() {
        return firstStoryModel;
    }

    public Boolean isFullyRead() {
        return fullyRead;
    }

    public void setFullyRead(final boolean fullyRead) {
        this.fullyRead = fullyRead;
    }
}

File: app/src/main/java/awais/instagrabber/models/FeedStoryModel.kt
package awais.instagrabber.models

import awais.instagrabber.repositories.responses.User
import awais.instagrabber.utils.TextUtils
import java.io.Serializable
import java.util.*

data class FeedStoryModel(
    val storyMediaId: String,
    val profileModel: User,
    var isFullyRead: Boolean,
    val timestamp: Long,
    val firstStoryModel: StoryModel?,
    val mediaCount: Int,
    val isLive: Boolean,
    val isBestie: Boolean
) : Serializable {
    val dateTime: String
        get() = TextUtils.epochSecondToString(timestamp)
}